### PR TITLE
publish: print on failure, simplify logic

### PR DESCRIPTION
**Pull request by hauntsaninja**
Should help with cases like #3397

The logic coalesces a couple things, so that we hit a single except clause
that prints "Uploading <package> failed"

The existing tests in https://github.com/python-poetry/poetry/blob/master/tests/publishing/test_uploader.py
should verify that this doesn't cause regressions.

### CODE REVIEW

**Comment by neersighted** [comment]
This is pretty stale now. Would you be willing to rebase onto current `master`?

**Comment by hauntsaninja** [comment]
Merged in master!

**Comment by neersighted** [comment]
This is blocked by `master`'s build being broken. I'd suggest rebasing onto #4753 -- this PR can be merged once that PR makes it into master.

**Review by neersighted — APPROVED** [review]

**Pull request merged**
